codenames_backend/routes/games.py

```diff
@@ -25,6 +29,92 @@ def new_game():
     db.session.commit()
     return game_schema.jsonify(game), 201
 
+def get_smallest_team(game_id):
+    teams = (
+        db.session.query(Player.team, func.count(Player.team))
+            .filter(Player.game_id == game_id)
+            .group_by(Player.team)
+            .order_by(func.count(Player.team))
+            .all()
+    )
+
+    if len(teams) == 1:
+        if teams[0][0] == "blue":
+            return "red"
+        else:
+            return "blue"
+    else:
+        return teams[0][0]
+
+
+@api.route("/join", methods=["POST"])
```

Comment:
I think this could be handled by landing the player in the "neutral" team when they join (or create) a game. Then they can select a team. If they are the first to join the team then they become the captain. There can be an auto-assign option if they choose that which does the smallest team thing (like counter-strike or assume most other team games)

If we move to the room model (with multiple consecutive games in each room) then it probably makes sense to select the captains on game start

Reply:
My thinking was that team colour doesn't really matter, so just auto-assign people to teams, but allow them to swap before the game starts. Simplifies jumping into the game, while still allowing choosing teams. Neutral team might make sense for spectators though... Will look into the room model.